pkg/internal/controller/controller.go

```diff
@@ -256,3 +264,11 @@ func (c *Controller) InjectFunc(f inject.Func) error {
 func (c *Controller) updateMetrics(reconcileTime time.Duration) {
 	ctrlmetrics.ReconcileTime.WithLabelValues(c.Name).Observe(reconcileTime.Seconds())
 }
+
+func (c *Controller) NeedLeaderElection() bool {
```

Comment:
Now that we implement this, I think this needs to return true by default, otherwise this is a fairly surprising change in default behavior (anything that doesn't implement LeaderElectionRunnable is assumed to use leader election to avoid breaking people too badly).

Reply:
(see below)